Tests/SonarScanner.MSBuild.PreProcessor.Test/RoslynAnalyzerProviderTests.cs

```diff
@@ -73,48 +76,176 @@ public void RoslynConfig_NoAssemblies()
+            {"sonaranalyzer-cs.staticResourceName", "SonarAnalyzer.zip"},
+            {"sonaranalyzer-cs.nuget.packageId", "SonarAnalyzer.CSharp"},
+            {"sonaranalyzer-cs.pluginVersion", "1.42.0"},
+            {"sonaranalyzer-cs.nuget.packageVersion", "1.13.0"},
+            {"sonaranalyzer-vbnet.analyzerId", "SonarAnalyzer.CSharp"},
+            {"sonaranalyzer-vbnet.ruleNamespace", "SonarAnalyzer.CSharp"},
+            {"sonaranalyzer-vbnet.pluginKey", "vbnet"},
+            {"sonaranalyzer-vbnet.staticResourceName", "SonarAnalyzer.zip"},
+            {"sonaranalyzer-vbnet.nuget.packageId", "SonarAnalyzer.CSharp"},
+            {"sonaranalyzer-vbnet.pluginVersion", "1.42.0"},
+            {"sonaranalyzer-vbnet.nuget.packageVersion", "1.13.0"},
+            {"sonar.cs.testPropertyPattern", "foo"},
+            {"sonar.sources", "**/*.*"},
+            {"sonar.cs.foo", "bar"},
+            {"sonar.vbnet.foo", "bar"},
+            {"sonar.cs.analyzer.security.pluginKey", "securitycsharpfrontend" },
+            {"sonar.cs.analyzer.security.pluginVersion", "1.42.0" },
+            {"sonar.cs.analyzer.security.staticResourceName", "SecurityAnalyzer.zip" },
+            {"sonaranalyzer.security.cs.pluginKey", "OLDSecurityCSharpFrontend" },
+            {"sonaranalyzer.security.cs.pluginVersion", "OLDSecurityCSharpFrontend" },
+            {"sonaranalyzer.security.cs.staticResourceName", "OLDSecurityCSharpFrontend" },
+        });
+        var context = new Context(TestContext, sonarProperties, [[@"c:\assembly1.dll"], [@"d:\foo\assembly2.dll"]], null, language);
+        var securityProperties = language == RoslynAnalyzerProvider.CSharpLanguage ?
+            """
+            
+                <Setting>
+                  <Key>sonar.cs.analyzer.security.pluginKey</Key>
+                  <Value>securitycsharpfrontend</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.cs.analyzer.security.pluginVersion</Key>
+                  <Value>1.42.0</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.cs.analyzer.security.staticResourceName</Key>
+                  <Value>SecurityAnalyzer.zip</Value>
+                </Setting>
+            """ : string.Empty;
+        var expectedSonarLintXml = $"""
+            <?xml version="1.0" encoding="UTF-8"?>
+            <AnalysisInput>
+              <Settings>
+                <Setting>
+                  <Key>sonar.{language}.analyzer.dotnet.pluginKey</Key>
+                  <Value>{language}</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.{language}.analyzer.dotnet.pluginVersion</Key>
+                  <Value>1.42.0</Value>
+                </Setting>
+                <Setting>
+                  <Key>sonar.{language}.analyzer.dotnet.staticResourceName</Key>
+                  <Value>SonarAnalyzer.zip</Value>
+                </Setting>
```

Comment:
Oversight from previous PRs: We shouldn't generate these to `SonarLint.xml` file, as `sonaranalyzer-{language}` where previously also not generated.

We use them to talk to scanner. We should not propagate them back to the analyzers